abstract.js

```diff
@@ -624,6 +624,7 @@ function abstractPersistence (opts) {
       var stream = instance.outgoingStream(client)
 
       stream.pipe(concat(function (list) {
+        list[0].messageId = updated.messageId
```

Comment:
I would prefer if we `delete` the `messageId` from `updated`, rather than vice versa.
also, can you add a check that they must not be the same object, but they must be equal?